piptools/utils.py

```diff
@@ -17,10 +21,33 @@ def format_requirement(ireq):
     Generic formatter for pretty printing InstallRequirements to the terminal
     in a less verbose way than using its `__str__` method.
     """
-    if ireq.editable:
-        line = '-e {}'.format(ireq.link)
+    if ireq.link and ireq.link.scheme in vcs.all_schemes:
+        # NOTE: might be called for debug logging, where the source_dir does
+        # not exist yet!
+        if ireq.source_dir and os.path.exists(ireq.source_dir):
+            line = get_src_requirement(dist=ireq.get_dist(),
+                                       location=ireq.source_dir,
+                                       find_tags=False)
+            line_link = Link(line)
+
+            # Prepend 'vcs+' which gets left out for 'git://', 'hg://' and
+            # 'bzr://' URLs.  Ref: https://github.com/pypa/pip/pull/2913.
+            if not '+' in line_link.scheme:
+                line = '{}+{}'.format(line_link.scheme, line)
```

Comment:
I really want to avoid reimplementing things that should be in pip. If this is currently not in pip, my judgment is we simply cannot use it. Please remove this hack and wait for this to be pulled into pip. I'm happy to bump the requirement to the then-latest pip version — no problem.